### PR TITLE
Add Tensorflow examples.

### PR DESCRIPTION
**Pull request by sile**
This PR adds the following two examples that use Tensorflow:
- examples/tensorflow_simple.py
- examples/pruning/tensorflow_integration.py

### CODE REVIEW

File: examples/tensorflow_simple.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+import numpy as np
+import shutil
+import tensorflow as tf
+
+import optuna
+
+MODEL_DIR = "/tmp/mnist_convnet_model"
+BATCH_SIZE = 128
+TRAIN_STEPS = 1000
```
Comment (toshihikoyanase) [inline examples/tensorflow_simple.py]: In my local environment, a study takes about 10 minutes, and it is about five times longer than the execution time of Chainer example. If we merge this PR, the execution time of CircleCI's example-pythonXX will be tripled.
Could you tune the parameters like training steps, batch size, and n_trials to reduce the execution time?

File: examples/tensorflow_simple.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+
+def create_network(trial, features):
+    # We optimize the numbers of layers and their units.
+    input_layer = tf.reshape(features['x'], [-1, 784])
+    prev_layer = input_layer
+
+    n_layers = trial.suggest_int('n_layers', 1, 3)
+    for i in range(n_layers):
+        n_units = int(trial.suggest_int('n_units_l{}'.format(i), 1, 128))
```
Comment (toshihikoyanase) [inline examples/tensorflow_simple.py]: This type conversion is redundant because `trial.suggest_int` returns integer values.

File: examples/tensorflow_simple.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        prev_layer = tf.layers.dense(inputs=prev_layer, units=n_units, activation=tf.nn.relu)
+
+    logits = tf.layers.dense(inputs=prev_layer, units=10, activation=tf.nn.relu)
```
Comment (toshihikoyanase) [inline examples/tensorflow_simple.py]: As far as I know, ReLU is not usually used to calculate logits. A dense layer without activation seems to be more suitable for examples.

c.f. Tensorflow examples
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/mnist/mnist.py#L82
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/layers/cnn_mnist.py#L87

File: examples/tensorflow_simple.py (hunk 2 above)
Comment (sile) [inline examples/tensorflow_simple.py]: Oops, sorry for my careless mistake...

File: examples/tensorflow_simple.py (hunk 3 above)
Comment (sile) [reply examples/tensorflow_simple.py]: You're right. I will fix it.

File: examples/tensorflow_simple.py (hunk 1 above)
Comment (sile) [reply examples/tensorflow_simple.py]: Okay. I will change `n_trials` to `25`. This value is the same as ChainerMN examples and seems sufficient to find out nearly optimal parameters for Tensorflow examples.

FYI
----

The followings are new execution times of examples on my Windows(WSL) laptop.
```console
$ time -p python3 examples/tensorflow_simple.py
Number of finished trials:  25

real 82.64
user 133.15
sys 62.09

$ time -p python3 examples/chainer_simple.py
Number of finished trials:  100

real 227.17
user 1006.57
sys 1062.59

$ time -p python3 examples/pruning/tensorflow_integration.py
  Number of finished trials:  25
  Number of pruned trials:  13
  Number of complete trials:  12

real 124.54
user 159.37
sys 48.15

$ time -p python3 examples/pruning/chainer_integration.py
Study statistics:
  Number of finished trials:  100
  Number of pruned trials:  90
  Number of complete trials:  10

real 128.71
user 495.78
sys 509.57
```

File: examples/tensorflow_simple.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+In this example, we optimize the validation accuracy of hand-written digit recognition using
+Tensorflow and MNIST. We optimize the neural network architecture as well as the optimizer
+configuration. As it is too time consuming to use the whole MNIST dataset, we here use a small
+subset of it.
+
+We have the following two ways to execute this example:
+
+(1) Execute this code directly.
+    $ python tensorflow_simple.py
```
Comment (g-votte) [inline examples/tensorflow_simple.py]: Because this example is specific to estimator interface, let's rename the filename to `tensorflow_estimator.py`.

File: examples/pruning/tensorflow_integration.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import shutil
+import tensorflow as tf
+
+import optuna
+
+MODEL_DIR = "/tmp/mnist_convnet_model"
```
[the remark this answers is not in the document]
Comment (sile) [reply examples/pruning/tensorflow_integration.py]: I basically agree with your opinion.
However, I have a concern. Since `tempfile.mkdtemp` does not clean up the created directory, if users execute this example many times, it will consume a huge capacity of their disk (over 700MB per execution).
Should we accept it? Or should we introduce a clean up mechanism?

[the remark this answers is not in the document]
Comment (sile) [reply examples/pruning/tensorflow_integration.py]: As a result of offline discussion, we decided to add `shutil.rmtree()` after `study.optimization()`.

[the remark this answers is not in the document]
Comment (sile) [reply examples/pruning/tensorflow_integration.py]: Done: https://github.com/pfnet/optuna/pull/323/commits/e84c158e254c974e7a4152d006b3558d50f72eac

File: examples/tensorflow_simple.py (hunk 4 above)
Comment (sile) [reply examples/tensorflow_simple.py]: We decided to rename `pruning/tensorflow_integration.py` to `pruning/tensorflow_estimator_integration.py`.

**Comment by sile** [comment]
@g-votte Thank you for your comments. I updated the example files.
